server/src/processor.rs

```diff
@@ -138,18 +141,32 @@ impl <'a> SignalProcessor<'a> {
         match signal.signal_status {
             SignalStatus::Settled => match self.current_settled.get_mut(&signal.account) {
                 Some(settled_signal) => {
-                    if signal.height > settled_signal.height {
+                    if signal.nonce > settled_signal.nonce {
```

Comment:
This is incorrect it should still check against the block height first - if the block heights are the same for both signals you should check against the nonce.